proxy/http/remap/PluginFactory.cc

```diff
@@ -94,6 +94,12 @@ PluginFactory::PluginFactory()
     }
   }
 
+  for (PluginRegInfo *plugin_info = globalPluginRegistry.head; plugin_info != nullptr; plugin_info = (plugin_info->link).next) {
```

Comment:
For this to work:
* `PluginRegInfo` had to be enhanced with `dso_reloading_enabled` flag
* `plugin_reg_list` had to be passed to `PluginFactory` constructor
* `PluginFactory` had to understand global `PluginRegInfo` internals. 
* `PluginFactory` would need to rebuild the local registry `optoutPlugins` on each config reload (by traversing `plugin_reg_list`)

May be we can simplify this design (avoid all of the above) and get rid of the unnecessary coupling with the following idea?

We could move the local `optoutPlugins` registry from `PluginFactory` into `LoadedPlugins`, a structure that already exists and survives the `PluginFactory` instances, which is what we actually need. 

We could add a method or methods to `LoadedPlugins` interface to manipulate the registry from within `TSPluginDSOReloadEnable` and then expose another `LoadedPlugins` method to be called from the `PluginFactory` to check if the plugin opted out from reloading when the remap plugin is instantiated.

@brbzull0, please feel free to ping me out-of-band, I will be glad to discuss!